### PR TITLE
resolved issue#21

### PR DESCRIPTION
**Pull request by mukultaneja**

### CODE REVIEW

**Review by dagwieers — CHANGES_REQUESTED** [review]
Please don't change anything except the needed changes. I do not agree with splitting lines. It does not make the code more readable, quite the opposite. So don't mix cosmetic changes (PEP8) with real modifications.

**Comment by sanand0** [comment]
Thanks @mukultaneja -- could you please introduce the test cases as discussed?

**Comment by mukultaneja** [comment]
Hi Anand, I have added a test case in the existing test_data method for Abdera Convention.

**Comment by sanand0** [comment]
@mukultaneja - the Travis checks are failing. Could you please take a look?

@mattgd - this is a PR aimed at the issue #21 you'd raised. Thought I'd loop you in.

**Comment by mattgd** [comment]
@sanand0 Thanks for the heads up, and @mukultaneja, thanks for fixing this issue. I appreciate it.

**Comment by mukultaneja** [comment]
@mattgd thanks.

**Comment by mukultaneja** [comment]
Hi @dagwieers,
I am looking into the issue that why Travis CI build is failing. After looking into the test report [https://travis-ci.org/sanand0/xmljson/jobs/247079647](https://travis-ci.org/sanand0/xmljson/jobs/247079647), I found that two existing test cases (1 for Abdera and 1 for Cobra) are failing, would you mind if I dig more into the code to resolve this build failing issue?

**Comment by dagwieers** [comment]
@mukultaneja The tests were working fine before. [Travis CI](https://travis-ci.org/sanand0/xmljson) They only started to fail with your commit, so I expect the failures to be caused by your changes. We will not accept your changes if it fails te Travis CI tests.

**Comment by mukultaneja** [comment]
thanks, @dagwieers I am looking into change and update it so that Travis CI tests will pass.

**Comment by mukultaneja** [comment]
Hi @sanand0 @dagwieers @mattgd,

I have updated the code for my pull request so that it could pass all the test cases. I have added more nested test cases as well to validate my new changes.

I would request all of you to take a look at the changes and the test cases once. Please let me know if they are correct or need some modifications.

File: xmljson/__init__.py [hunk 1]
```diff
@@ -223,16 +223,10 @@ def data(self, root):
                     children_list = [self._fromstring(text), ]
 
         count = Counter(child.tag for child in children)
```
Comment (dagwieers) [inline xmljson/__init__.py]: If your code is correct, `count` here is also not needed.

**Review by dagwieers — CHANGES_REQUESTED** [review]

File: xmljson/__init__.py (hunk 1 above)
Comment (mukultaneja) [reply xmljson/__init__.py]: HI @dagwieers, thanks for the suggestion. I have removed the count variable and updated the code.

**Comment by dagwieers** [comment]
I would like to get feedback from @sanand0 as he wrote the original code.

**Comment by sanand0** [comment]
I'm perfectly fine with this @dagwieers - this affects only the Abdera convention that you'd authored, so if you're OK with the test cases and the change, I'm perfectly happy to merge. It's a minimalist change, so that's great too.

Are the revised test cases in line with your expectation on the Abdera convention?

**Pull request merged**